File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int maxif=0;
        int freq[]=new int[26];
        int maxil=0;
        int l=0;
        int n=s.length();
        for(int r=0;r<n;r++)
        {
            int x=s.charAt(r)-'A';
            freq[x]++;
            maxif=Math.max(maxif,freq[x]);
            while((r-l+1)-maxif>k)
            {
                freq[s.charAt(l)-'A']--;
                l++;
            }
            maxil=Math.max(maxil,(r-l+1));
        }
        return maxil;
    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.length();
        int ans = 0;
        for(char ch = 'A' ; ch <= 'Z'; ch++){
            queue<int>q;
            int i = 0, j = 0;

            while(j < n){
                if(s[j] != ch) q.push(j);
                if(q.size() > k){
                    i = q.front() + 1;
                    q.pop();
                }
                ans = max(ans , j - i + 1);
                j++;
            }
        }
        return ans;
    }
};